### PR TITLE
Updated tests from tensorflow.keras.layers.convolutional_recurrent_test

### PR DESCRIPTION
**Pull request by willgraf**
Parameterized the `test_conv_gru_2d` test.

Added several new tests from the TensorFlow tests including:
* test_conv_gru_2d_statefulness
* test_conv_gru_2d_regularizers
* test_conv_gru_2d_dropout
* test_conv_gru_2d_cloning

Fixed a potential bug with dropout where we were passing `h_tm1` instead of `states[1]`.

Also some general linting fixes.

### CODE REVIEW

File: deepcell/layers/convolutional_recurrent.py [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+                                        return_sequences=return_sequences,
+                                        return_state=return_state,
+                                        go_backwards=go_backwards,
+                                        stateful=stateful,
+                                        **kwargs)
+
+        self.activity_regularizer = regularizers.get(activity_regularizer)
+
+    def call(self, inputs, mask=None, training=None, initial_state=None):
+        # self.cell.reset_dropout_mask()
```
Comment (willgraf) [inline deepcell/layers/convolutional_recurrent.py]: Should we reset the masks as happens in the ConvLSTM2d?

**Comment by willgraf** [comment]
merge was done outside a PR, this PR does not serve a purpose.
